codalab/codalab-worksheets
# Respect .gitignore when archiving

Fixes #1312 

Changes:
- When archiving, ignore entries specified by `.gitignore` by default. It also works for `.gitignore`'s in subdirectories.
- Exclude `.git/` always
- Added `-i` option for `cl upload` where the user can specify a different ignore file.

How to test:
- `cl upload some_directory` where `some_directory` has `.gitignore`.
- `cl upload some_directory -i='.tarignore'`  should ignore `.tarignore`.
- `cl upload some_directory -i=''`  shouldn't ignore.

I also added a test that covers this logic in `file_util_test.py`.
